department-of-veterans-affairs/caseflow
# Ryan appeals/17513 redo

Resolves #((https://vajira.max.gov/browse/APPEALS-17513)

# Description
Pre-populates the MST/PACT checkbox on the add issues page if the data is present in the model. 

This is attempt #2 with RadioField.jsx reverted and the new IntakeRadioField.jsx holding the correct info. This also includes a new method for mst_available and pact_available that I worked out with Manesh.

**contestable_issue.rb**
```
 # cycle the issues to see if the past decision had any mst codes on contentions
  def mst_available?
    source_request_issues&.each do |issue|
      return contested_by_request_issue&.mst_contention_status?
    end
  end

  # cycle the issues to see if the past decision had any pact codes on contentions
  def pact_available?
    source_request_issues&.each do |issue|
      return contested_by_request_issue&.pact_contention_status?
    end
  end
```

## Acceptance Criteria

## Testing Plan
1. Go to Intake and use one of the following veteran numbers. (This may be blocked by new seed data that's coming soon)
- 992190636
- 107924012
- 329146737
- 981775045
- 738464961
- 245620198
- 955197769
2. Fill out the form with any options until you reach the add issues page.
3. Click the Add Issues button and select any of the listed issues.
4. Observe that you cannot change the value in the checkboxes and that they are already checked.


# Frontend
## User Facing Changes

WIREFRAME LINK
https://www.figma.com/proto/va7aL4f0BPdh5LcPO0I8O6/PI-12-(2023)%3A-Intake?node-id=104-20554&scaling=min-zoom&page-id=0%3A42964&starting-point-node-id=0%3A45310&show-proto-sidebar=1

 BEFORE|AFTER
 ---|---
<img width="751" alt="disabled autopop" src="https://user-images.githubusercontent.com/110805785/235526978-495fedd8-1a72-456b-9083-adac9839afe0.png">
<img width="747" alt="enabled autopop" src="https://user-images.githubusercontent.com/110805785/235526998-c9da9d4b-cf59-4f5f-9689-a18c75dadb98.png">


## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:
